rules/aep0004/resource_pattern.go

```diff
@@ -36,9 +37,18 @@ var resourcePattern = &lint.MessageRule{
 	},
 }
 
-func lintResourcePattern(resource *annotations.ResourceDescriptor, desc desc.Descriptor, loc *dpb.SourceCodeInfo_Location) []lint.Problem {
+func lintResourcePattern(resource *aepapi.ResourceDescriptor, desc desc.Descriptor, loc *dpb.SourceCodeInfo_Location) []lint.Problem {
+	return lintResourcePatternCommon(resource.GetPattern(), desc, loc)
+}
+
+// lintResourcePatternGoogleAPI lints resource patterns for Google API ResourceDescriptor (used for file-level resource definitions)
```

Comment:
should we be allowing file-level resource definitions at all? I'm unclear where this pattern would be valuable.

If it was valuable, I think we should just add a `aep.api.file_info` annotation similar to `aep.api.field_info`?